src/components/App.js

```diff
@@ -1,18 +1,212 @@
-import React from 'react';
+import React from "react";
 
-import '../styles/App.scss';
+import "../styles/App.scss";
+import Menu from "./Menu";
+import Order from "./Order";
+import OrderHistory from "./OrderHistory";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faShoppingCart, faHome } from "@fortawesome/free-solid-svg-icons";
+
+library.add(faShoppingCart, faHome);
 
 class App extends React.Component {
-  constructor(){
+  constructor() {
     super();
+    this.state = {
+      menu: [],
+      order: [],
+      placedOrders: [],
+      orderBasket: [],
+      displayMenu: true,
+      displayOrderHistory: false
+    };
+
+    this.receiveAddClick = this.receiveAddClick.bind(this);
+    this.receiveRemoveClick = this.receiveRemoveClick.bind(this);
+    this.receiveClickPlus = this.receiveClickPlus.bind(this);
+    this.receiveClickMinus = this.receiveClickMinus.bind(this);
+    this.receiveOrderSubmit = this.receiveOrderSubmit.bind(this);
+    this.shoppingCart = this.shoppingCart.bind(this);
+    this.orderHistory = this.orderHistory.bind(this);
+  }
+
+  componentDidMount() {
+    fetch("/api/menu")
+      .then(response => response.json())
+      .then(body => {
+        const menu = Object.values(body);
+
+        this.setState({
+          menu: menu
+        });
+      });
+
+    const storageString = window.localStorage.getItem("placedOrders");
+    const storage = !storageString ? [] : JSON.parse(storageString);
+    const orderString = window.localStorage.getItem("order");
+    const orderStorage = !orderString ? [] : JSON.parse(orderString);
+    this.setState({
+      placedOrders: storage,
+      order: orderStorage
+    });
+  }
+
+  receiveOrderSubmit() {
+    fetch("/api/order", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json; charset=utf-8"
+      },
+
+      body: JSON.stringify(this.state.order)
+    })
+      .then(response => response.json())
+      .then(body => {
+        this.setState({
+          order: [],
+          displayMenu: true
+        });
+        alert("Order has been placed successfully!");
+      });
+
+    fetch("/api/order")
+      .then(response => response.json())
+      .then(body => {
+        this.setState(
+          {
+            placedOrders: body
+          },
+          () =>
+            localStorage.setItem(
```

Comment:
Since placedOrders are stored on the server, I would suggest not storing them locally as well. Otherwise we end up with 2 sources of truth. It would better to fetch the data from the server when needed